### PR TITLE
feat(subgraph) - receipt numbers

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description

Add `receiptNumber` to subgraph  

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Review by julien51 — CHANGES_REQUESTED** [review]
This is not right. The receipt number is not the number of keys... A receipt could exist for a transaction with multiple keys and a receipt could exist for a renewals (no new key).

File: subgraph/schema.graphql [hunk 1]
```diff
@@ -31,6 +31,8 @@ type Lock @entity {
   lastKeyMintedAt: BigInt
   "Address of the lock deployer"
   deployer: Bytes!
+  "Receipts number of lock"
+  receiptCount: BigInt!
```
Comment (julien51) [inline subgraph/schema.graphql:35]: This is wrong. How can you keep track of the receipt count if you don't have a way to count them??

[the remark this answers is not in the document]
Comment (kalidiagne) [reply subgraph/schema.graphql:35]: we can rename to `numberOfReceipt` or add `receipts` and treat it like the `keys` as you suggested eventually

Comment (julien51) [inline subgraph/schema.graphql:35]:
```suggestion
  numberOfReceipts: BigInt!
```

File: subgraph/schema.graphql [hunk 2]
```diff
@@ -112,4 +114,6 @@ type Receipt @entity {
   amountTransferred: BigInt
   "Total gas paid"
   gasTotal: BigInt!
+  "Increasing number of receipt"
+  receiptNumber: BigInt
```
Comment (julien51) [inline subgraph/schema.graphql:118]:
```suggestion
  numberOfReceipts: BigInt
```

File: subgraph/src/public-lock.ts [hunk 3]
```diff
@@ -468,5 +468,13 @@ export function createReceipt(event: ethereum.Event): void {
   receipt.sender = event.transaction.from.toHexString()
   receipt.tokenAddress = tokenAddress.toHexString()
   receipt.gasTotal = BigInt.fromString(totalGas.toString())
+
+  if (lock) {
+    const newReceiptNumber = lock.receiptCount.plus(BigInt.fromI32(1))
+    lock.receiptCount = newReceiptNumber
```
Comment (julien51) [inline subgraph/src/public-lock.ts:474]:
```suggestion
    lock.numberOfReceipts = newReceiptNumber
```

File: subgraph/schema.graphql (hunk 1 above)
[the remark this answers is not in the document]
Comment (julien51) [reply subgraph/schema.graphql:35]: Ok let's do `numberOfReceipts` (with an `s`) and then it's good!

**Review by julien51 — APPROVED** [review]
LG!

File: subgraph/tests/locks.test.ts [hunk 4]
```diff
@@ -72,6 +72,7 @@ describe('Describe Locks events', () => {
     assert.fieldEquals('Lock', lockAddress, 'tokenAddress', nullAddress)
     assert.fieldEquals('Lock', lockAddress, 'lockManagers', `[]`)
     assert.fieldEquals('Lock', lockAddress, 'totalKeys', '0')
+    assert.fieldEquals('Lock', lockAddress, 'receiptCount', '0')
```
Comment (kalidiagne) [inline subgraph/tests/locks.test.ts:75]: @julien51 awesome so ill update also test with `numberOfReceipts `

**Pull request merged**